iceoryx2-bb/cxx/include/iox2/container/static_string.hpp

```diff
@@ -454,6 +500,22 @@ class StaticString {
         return !(lhs == rhs);
     }
 
+    friend auto operator<(StaticString const& lhs, StaticString const& rhs) -> bool {
+        return lhs.compare(rhs) < 0;
+    }
```

Comment:
A not-so-obvious property of the UTF-8 encoding is that this actually orders strings lexicographically _by code points_. This may or may not be worth pointing out in a comment, but I think it's pretty cool.